src/parse.ts

```diff
@@ -947,6 +968,33 @@ function parseMediaPlaylist(lines, params) {
     if (playlist.lowLatencyCompatibility) {
         checkLowLatencyCompatibility(playlist, containsParts);
     }
+
+    let transformedSegments: Array<Segment> = playlist.segments;
+    // Apply any user-specified Segment transformations
+    // Ideally we'd use map(), but we want the transformers to be able to remove segments
+    params.segmentTransformers?.forEach((transformerFunc: SegmentTransformer) => {
```

Comment:
Instead of iterating after - would it be possible to run the segments as they come in (line 955)? There may be side-effects when the initial processing finishes, for example the `checkDateRange` function runs before this transformer.

Reply:
Good idea- I'll move it on up